### ISSUE TITLE
Class and Namespace Name Collision

### ISSUE THREAD

**Issue by lawrence-laz** [swharden/Spectrogram#30]
Main class and namespace in which it is declared share the same name.
```csharp
namespace Spectrogram
{
    public class Spectrogram
    {
...
```

This causes some inconvenience when creating an instance.
```csharp
using Spectrogram;

var spec = new Spectrogram(...);
// CS0118: 'Spectrogram' is a namespace but is used like a type
```

I know this is a breaking change, but have you considered changing the namespace's name?

**Comment by swharden** [comment]
Hi @lawrence-laz, thanks for mentioning this! I hadn't considered this problem before. I never add the `using` to my code and seeing `Spectrogram.Spectrogram()` never bothered me, but now that you pointed it out it's all I see! I wish I could go back in a time machine and name the namespace (and NuGet package) something different.

## Problem and Immediate Solution

```cs
// Causes error CS0118: 'Spectrogram' is a namespace but is used like a type
using Spectrogram;
var spec = new Spectrogram(...);
```

```cs
// This works, but is long and looks redundant
var spec = new Spectrogram.Spectrogram(...);
```

```cs
// This works, but is nontraditional
using SG = Spectrogram;
var spec = new SG.Spectrogram(...);
```

## Potential Fix: Rename the Namespace

> have you considered changing the namespace's name?

I could rename the `Spectrogram` namespace, but this would be a very hard break for any existing code. The NuGet package is named `Spectrogram` too, so I think the namespace should probably stay as it is.

## Potential Fix: Rename the Class

The `Spectrogram` class could be renamed though. The old one could be left and marked obsolete so no one's existing code would break. The problem with this is that it's hard to think of a reasonable name to describe that class...
  * `SGram` <-- I think I'm preferring this
  * `SpGram`
  * `Specgram`
  * `Spectrogram2D`
  * `SpectrogramImage`
  * `SpectrogramGenerator`
  * `SpectrogramProcessor`

The proposed change would allow user code to look like this:

```cs
using Spectrogram;
var spec = new SGram(...);
```

**Comment by swharden** [comment]
I created a PR (#31) where the class is renamed. I'll probably let this sit here for a few days in case anyone wants to say anything about it before I merge it in and release a new package on NuGet. Using the old `Spectrogram` class should still work, but it will cause an obsolete warning. Hopefully it won't ruffle too many feathers 🤞 

https://github.com/swharden/Spectrogram/blob/0f73874fd036235cd8e0ac8da589bcb34b5d6ab7/src/Spectrogram/Obsolete.cs#L7-L14

**Comment by lawrence-laz** [comment]
Renaming a class seems like a very good alternative. 👍

The name `SGram` however strikes as a bit odd and cryptic. It's unusual for me to see shortened names in c#, but I guess this is subjective. To me the name `SpectrogramGenerator` seemed a bit better and self-descriptive, given that this class takes some inputs and then generates images, which are the actual spectrograms.

**Comment by swharden** [comment]
> name SGram however strikes as a bit odd and cryptic

> name SpectrogramGenerator seemed a bit better

I think I agree with you here!

```cs
// this is shorter, but cryptic
var sg = new SGram(sampleRate, fftSize: 4096, stepSize: 500, maxFreq: 3000);
sg.Add(myData);
sg.SaveImage("data.png");
```

```cs
// this is more to type, but more expressive
var sg = new SpectrogramGenerator(sampleRate, fftSize: 4096, stepSize: 500, maxFreq: 3000);
sg.Add(myData);
sg.SaveImage("data.png");
```

Expressiveness and readability should be favored over character count...

I'll sleep on this one for a day or two and come back with fresh eyes before I make a final call! Thanks again for pointing it out @lawrence-laz. I'll leave this issue open until I fix it and the new package is in NuGet

**Comment by swharden** [comment]
Thanks again for raising this issue @lawrence-laz! I ended-up changing it to `SpectrogramGenerator`, updated all the documentation, and released version `1.3.0` on NuGet 👍
